File: partitionLabels_763.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        if not s:
            return ''
        ranges=[]
        letters=set(x for x in s)
        for letter in letters:
            left = s.find(letter)
            right = s.rfind(letter)
            ranges.append([left, right])
        ranges.sort()
        [left,right] = ranges[0]
        result=[]

        for i in range(1,len(ranges)):
            if ranges[i][0]>right:
                result.append(right-left+1)
                [left,right] = ranges[i]
            else:
                right = max(right,ranges[i][1])

        result.append(right-left+1)
        return result

File: partitionLabels_763.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
    int last[26];
    for(int i=0;i<s.size();++i){
        last[s[i]-'a']=i;
    }
    int start=0;
    int end=0;
    std::vector<int> partition;
    for(int i=0;i<s.size();++i){
        end=std::max(end,last[s[i]-'a']);
        if(end==i){
            partition.push_back(end-start+1);
            start=end+1;
        }
    }
    return partition;
    }
};
